accepted/2024/repo-level-lock-file.md

```diff
@@ -0,0 +1,84 @@
+﻿# ***Repository Level Lock File***
+
+- Author: [@CEbbinghaus](https://github.com/CEbbinghaus) 
+- Issue: [Nuget/Home#12409](https://github.com/NuGet/Home/issues/12409) 
+
+## Summary
+
+Applications consisting of multiple solutions (using assembly references between projects) cannot control their transitive dependencies by using Nuget. The current lock file is on a per-project basis rather than an application/repository basis. Given a `Directory.packages.props` file, each project could resolve different versions of transitive dependencies, which a single central lock file would solve by allowing consistency between package versions across multiple solutions.
+
+
+## Motivation 
+
+Monorepositories with multiple solutions utilizing assembly references that deploy all their code together as a single application are unable to use Nuget (with or without CPM) due to transitive version conflicts. This is due to NuGet being able to resolve different versions of transitive dependencies for each individual build, as they occur in isolation from one another. A single central lock file allows all transitive dependencies to have locked versions defined at the root, thereby eliminating third-party dependency management software such as Paket.
+
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented, and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real-life examples, and low-fidelity mockups or pseudocode to show how this proposal would look. -->
+
+The `Directory.packages.props` now uses the`<NuGetLockFilePath>[path]</NuGetLockFilePath>` property within a `<PropertyGroup>` to specify its lock file path. It becomes the source of truth for all builds. Project level lock files are no longer required unless a package version is overridden.
```

Comment:
There's a technical complexity here since a property value doesn't get path resolved, so knowing where it comes would be required. 
The simpler approach might be the already mentioned just making the file live next to the directory.packages.props.

Reply:
For flexibility I prefer the idea of having the lock file path configurable which also tracks consistently with lock files on a project. The better solution is probably setting the default property to be next to the `Directory.packages.props` and instead offering (if it doesn't already exist) a `$(ThisCurrentFilePath)` MSBuild property which can be used to set the file path relative to the Directory.packages.props rather than something relative to the project.

Reply:
Yeah, it's all doable, we'd just need to be specific to avoid confusion.